### PR TITLE
Adaptive and concurrent pruning code cleanup

### PR DESCRIPTION
**Pull request by mdarade**
* **Problem Overview**  
This change cleans up code of following sub-components spread across system.
Ticks Infra (used in pruning context only) - Used to reconfigure Blocks/Second to prune at every new tick.
Reserved pages client - Used to store newly agreed upon Blocks/Second pruning rate.
Resource Manager - Used to monitor resource consumption in order to decide pruning speed.
Concurrent pruning - Pruning running in async manner during ongoing traffic.
Adaptive Pruning - Parent object composing all above mentioned elements.

* **Testing Done**  
Executed all tests in local environment

### CODE REVIEW

**Comment by mdarade** [comment]
> * Ticks Infra (used in pruning context only)

Core ticks generator infra is being used at other places too so we are keeping it.
I've just removed pruning code which was making use of ticks infra to generate ticks at specific interval to review blocks/sec to prune.


Same holds true for reserved pages.
There are many clients of reserved pages.
I've removed only pruning code which was storing agreement (batch-blocks-num etc) using reserved pages.

File: reconfiguration/cmf/concord.cmf [hunk 1]
```diff
@@ -414,28 +410,6 @@ Msg PruneLegacyConfiguration 78 {
   uint32 tick_period_seconds
```
Comment (vrancurel) [inline reconfiguration/cmf/concord.cmf:410]: Why do we need to keep this ?

Comment (mdarade) [reply reconfiguration/cmf/concord.cmf:410]: PruneRequest has number of fields out of which I've kept these 2 even though they are not required anymore.

 # TicksGenerator period in seconds.
  uint32 tick_period_seconds

  # The number of blocks in a pruning batch.
  uint64 batch_blocks_num 
  
  
  I am planning to do cleanup these in next PR.

**Review by vrancurel — APPROVED** [review]

File: kvbc/CMakeLists.txt [hunk 2]
```diff
@@ -19,7 +19,6 @@ add_library(kvbc  src/ClientImp.cpp
     src/direct_kv_storage_factory.cpp
     src/merkle_tree_storage_factory.cpp
     src/pruning_handler.cpp
-    src/pruning_reserved_pages_client.cpp
```
Comment (cloudnoize) [inline kvbc/CMakeLists.txt:22]: This PR removes adaptive pruning, is this file also relevant for the concurrent pruning?

File: kvbc/cmf/pruning_msgs.cmf [hunk 3]
```diff
@@ -1,17 +0,0 @@
-# Represents a pruning agreement, i.e. when processing a PruneRequest.
-Msg Agreement 1 {
```
Comment (cloudnoize) [inline kvbc/cmf/pruning_msgs.cmf:2]: Same as above, is it ok to remove it without removing concurrent turning?

Comment (mdarade) [reply kvbc/cmf/pruning_msgs.cmf:2]: PR is also going to remove concurrent pruning code. I've updated title & description of PR.

File: kvbc/CMakeLists.txt [hunk 4]
```diff
@@ -64,7 +60,7 @@ if (BUILD_ROCKSDB_STORAGE)
 
 endif (BUILD_ROCKSDB_STORAGE)
 target_link_libraries(kvbc PUBLIC corebft util)
-target_link_libraries(kvbc PUBLIC categorized_kvbc_msgs pruning_msgs event_group_msgs)
```
[the remark this answers is not in the document]
Comment (mdarade) [reply kvbc/CMakeLists.txt:67]: PR is also going to remove concurrent pruning code. I've updated title & description of PR.

File: kvbc/CMakeLists.txt (hunk 2 above)
Comment (mdarade) [reply kvbc/CMakeLists.txt:22]: PR is also going to remove concurrent pruning code. I've updated title & description of PR.

**Review by cloudnoize — APPROVED** [review]
That's great, thanks!

**Pull request merged**